website/static/js/formViewModel.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('js/osfHelpers');
+var oop = require('js/oop');
+
+var ValidationError = oop.extend(Error, {
+    constructor: function (messages, header, level) {
+        this.super.constructor.call(this);
+        this.messages = messages || [];
+        this.header = header || 'Error';
+        this.level = level || 'warning';
+    }
+});
+
+var FormViewModel = oop.defclass({
+    /**
+    * Base class KO viewmodel based forms should inherit from.
+    *
```

Comment:
Not really an ES thing; more of a convention. [YUIDoc](https://yui.github.io/yuidoc/) and most other doc styles follow this.